### PR TITLE
backport: Merge bitcoin#18842, 21162, 21043, 17556, 18017, 8795, 19362, 20932, bitcoin-core/gui#162

### PR DESCRIPTION
**Pull request by vijaydasmp**
## Issue being fixed or feature implemented
bitcoin backports

## What was done?
Backported bitcoin changes

## How Has This Been Tested?
CI Run

### CODE REVIEW

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 , requesting review

**Review by knst — COMMENTED** [review]
Please rename a commit `Merge #162: Add network to peers window and peer details`
to
`Merge bitcoin-core/gui#162: Add network to peers window and peer details`

to avoid confuction with #162 (due to wrong github UI resolving to this link) and to avoid possible confusion while review/looking to history.

**Review by knst — DISMISSED** [review]
utACK

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 @PastaPastaPasta , requesting review

File: src/test/fs_tests.cpp [hunk 1]
```diff
@@ -68,6 +68,23 @@ BOOST_AUTO_TEST_CASE(fsbridge_fstream)
...
+        BOOST_CHECK_EQUAL(tmpfile2, p);
+    }
+    {
+        // Ensure joining with empty paths does not add trailing path components.
+        BOOST_CHECK_EQUAL(tmpfile1, fsbridge::AbsPathJoin(tmpfile1, ""));
+        BOOST_CHECK_EQUAL(tmpfile1, fsbridge::AbsPathJoin(tmpfile1, {}));
+    }
 }
 
 BOOST_AUTO_TEST_SUITE_END()
```
Comment (UdjinM6) [inline src/test/fs_tests.cpp:90]: 20932: newline at eof
```suggestion
BOOST_AUTO_TEST_SUITE_END()

```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
18842: we don't have rbf so 18842 should not be partial (not skipping it because this code will be used later in future backports outside of rbf)

**Review by UdjinM6 — APPROVED** [review]
utACK

**Comment by vijaydasmp** [comment]
Hello @knst @UdjinM6 and @PastaPastaPasta , Requesting review again, had to fix merge conflict, on src/addrman.h file ,

**Review by knst — APPROVED** [review]
utACK

note for reviewers: bitcoin/bitcoin#18722 is not partial due to prior backporting in https://github.com/dashpay/dash/pull/5681

**Comment by UdjinM6** [comment]
also, no commit is marked "partial" now so PR title shouldn't have "(Partial)" either

**Comment by vijaydasmp** [comment]
> [ff5dcbe](https://github.com/dashpay/dash/commit/ff5dcbe4db128b328703eef52fce7fdce438d05f) is meaningless, should be dropped

dropped 18722

**Review by UdjinM6 — APPROVED** [review]
re-utACK

**Review by PastaPastaPasta — APPROVED** [review]
utACK for merging via merge commit

**Comment by UdjinM6** [comment]
ping @knst

**Pull request merged**